### PR TITLE
React Calculator: Milestone 4-logic

### PR DESCRIPTION
**Pull request by ferbaco86**
In this branch, I added the modules calculate and operate for dealing with the logic behind the calculator. 

## MIlestone Requirements

- [x] Create a new logic folder inside src and initialize two javascript files there: calculate.js and operate.js.
- [x] Add the big.js package to your project dependencies.
- [x] Implement the calculate.js module.
- [x] Implement the operate.js module. 
- [x] Add calculate.js to the App component.

### CODE REVIEW

File: src/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import operate from './operate';
+
+const calculate = ({ total, next, operation }, buttonName) => {
+  const calculator = {
+    total,
+    next,
+    operation,
+  };
```
Comment (bellom) [inline src/logic/calculate.js]: According to the instructions on the pathwright, we need to pass an object as an argument.

The argument passed to the `function calculate` should be an object, we can achieve this by using destructuring.
```
function calculate(data, buttonName) {
  let { total, next, operation } = data;
.....
}
```

**Quick one:** ` const calculator = {` You should be using `let` cause the value would be changing.

image

https://microverse.pathwright.com/library/react-redux-v11/184438/path/step/106376018/

File: src/logic/operate.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      break;
+    case '÷':
+      if (valueTwo === 0) {
+        total = "Can't divide by zero";
+      } else {
+        total = valueOne.div(valueTwo);
+      }
+      break;
+    case '%':
+      total = valueOne.div(100);
```
Comment (bellom) [inline src/logic/operate.js:26]: ### [Optional]
Job well done but you don't have to do the calculating yourself, that where the use of `big` comes in -> [big js library](http://mikemcl.github.io/big.js/#mod)

Example: case `%` will be -> `somevalue.mod(somevalue)`

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @ferbaco86  :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

File: src/logic/operate.js (hunk 2 above)
Comment (ferbaco86) [reply src/logic/operate.js:26]: Thanks for the tip, but I think, in this case, we are not looking for the remainder, or modulo, instead this symbol in the calculator means the percentage of a number.

File: src/logic/calculate.js (hunk 1 above)
Comment (ferbaco86) [reply src/logic/calculate.js]: Thanks, changes applied.

File: src/logic/operate.js (hunk 2 above)
Comment (bellom) [reply src/logic/operate.js:26]: @ferbaco86 Yes, we are not looking remainder, in this case, you only supply the first digit right?
`example: click `10` then `%` = 0.01
`

**Review by bellom — APPROVED** [review]
## Hi @ferbaco86  :clap: 

**Great work making all the necessary corrections from the previous review.**

# Status: Approved 👍 

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

File: src/logic/operate.js (hunk 2 above)
Comment (ferbaco86) [reply src/logic/operate.js:26]: Yes, only the first one

**Pull request merged**